azext_iot/iothub/providers/device_stream.py

```diff
@@ -0,0 +1,36 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from typing import Optional
+from knack.log import get_logger
+from azext_iot.common.embedded_cli import EmbeddedCLI
+from azext_iot.iothub.providers.base import IoTHubProvider
+from azext_iot.iothub.common import HUB_PROVIDER
+
+logger = get_logger(__name__)
+DEVICE_STREAMS_API_VERSION = "2023-06-30-preview"
+
+
+class DeviceStreamProvider(IoTHubProvider):
+    def __init__(
+        self,
+        cmd,
+        hub_name: str,
+        rg: Optional[str] = None,
+    ):
+        super(DeviceStreamProvider, self).__init__(cmd, hub_name, rg)
+        self.cli = EmbeddedCLI(cli_ctx=self.cmd.cli_ctx)
+
+    def show(self) -> dict:
+        result = self.cli.invoke(
+            f"resource show -n {self.target['name']} -g {self.target['resourcegroup']} "
+            f"--api-version {DEVICE_STREAMS_API_VERSION} --resource-type {HUB_PROVIDER}",
+            capture_stderr=True
```

Comment:
What is the intended behavior of the command if an API error happens (looking at capture_stderr=True)?

Reply:
This was the old cli - which I do want to rewrite a bit but it would be a bigger task

The capture_stderr=True results in the error just being actually raised rather than just printed out in console. By having it raised, the error should present better (rather than be a json decode error)

Reply:
Ok. It's been a long while since I worked on the code base and forgot the capture_stderr dynamic, a more appropriate name for it to me would be something like "raise_stderr".